podman/domain/config.py

```diff
@@ -62,17 +67,43 @@ class PodmanConfig:
     def __init__(self, path: Optional[str] = None):
         """Read Podman configuration from users XDG_CONFIG_HOME."""
 
+        self.is_default = False
         if path is None:
             home = Path(xdg.BaseDirectory.xdg_config_home)
-            self.path = home / "containers" / "containers.conf"
+            self.path = home / "containers" / "podman-connections.json"
+            old_toml_file = home / "containers" / "containers.conf"
+            self.is_default = True
+        # this elif is only for testing purposes
+        elif "@@is_test@@" in path :
```

Comment:
So I tried using mock env to imitate the path we default to but it just wouldn't work. This is the best work around I could come up with. If anyone is able to figure out the mock env thing I am happy to remove this and update the test accordingly.

Looking at the code, seems like we grab the path to look at based on `XDG_CONFIG_HOME` and that is what I kept trying to override in the tests, but it would always end up using the path on my local machine instead.